include/alpaka/atomic/Traits.hpp

```diff
@@ -80,6 +82,41 @@ namespace alpaka
                     value);
         }
 
+        //-----------------------------------------------------------------------------
+        //! Executes the given operation atomically.
+        //!
+        //! \tparam TOp The operation type.
+        //! \tparam TAtomic The atomic implementation type.
+        //! \tparam T The value type.
+        //! \param atomic The atomic implementation.
+        //! \param addr The value to change atomically.
+        //! \param compare The comparison value used in the atomic operation.
+        //! \param value The value used in the atomic operation.
+        //-----------------------------------------------------------------------------
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TOp,
+            typename TAtomic,
+            typename T>
+        ALPAKA_FN_HOST_ACC auto atomicOp(
```

Comment:
My point goes in the direction that not all atomics had a address, compare and value.
But I am fine with this implementation if this is still valid for all alpaka provided atomic.